packages/terminal/src/browser/terminal-frontend-contribution.ts

```diff
@@ -120,6 +120,7 @@ export class TerminalFrontendContribution implements TerminalService, CommandCon
     @postConstruct()
     protected init(): void {
         this.shell.currentChanged.connect(() => this.updateCurrentTerminal());
+        this.shell.currentChanged.disconnect(() => this.updateCurrentTerminal());
```

Comment:
What do we want to achieve here? Right now it create a new function here, just to disconnect it. Does not seem to have any effect.

Reply:
It's a phosphor API for event listening: `connect` == subscribe and `disconnect` === unsubscribe

It would be good to deprecate and replace with our events instead. Unfortunately, it's heavily used.

Reply:
`shell.currentChanged` is fired when a last focused widget is changed